### PR TITLE
CORE-53571 use regular object prototype for config objects

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

We were logging an object created with Object.create(null) which caused problems on a Shopify site where the logging was overwritten. This change fixes that case and adds a functional test to make sure it doesn't happen again.

## Related Issue

https://github.com/adobe/alloy/issues/611
https://jira.corp.adobe.com/browse/CORE-53571

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: test/functional/specs/Logging/C532204.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  title: "C532204: Logged objects can be stringified"
+});
+
+test.meta({
+  ID: "C532204",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const setupLogger = ClientFunction(() => {
```
Comment (liljenback) [inline test/functional/specs/Logging/C532204.js:25]: This will cover all scenarios we need from the `triggerAlloyEvent` output?

**Review by liljenback — APPROVED** [review]

**Review by Aaronius — APPROVED** [review]
Thanks @jonsnyder. This looks good. I would recommend adding a comment to the functional test explaining why it's important that logged information can be stringified, lest the reasoning be long to the sands of time.

File: test/functional/specs/Logging/C532204.js (hunk 1 above)
Comment (jonsnyder) [reply test/functional/specs/Logging/C532204.js:25]: Actually, simply configuring alloy reproduced the bug, but triggering an alloy event creates a lot of other logging statements. This test doesn't exercise all the logging scenarios, but I felt like it was good enough.

**Comment by jfkhoury** [comment]
> Thanks @jonsnyder. This looks good. I would recommend adding a comment to the functional test explaining why it's important that logged information can be stringified, lest the reasoning be lost to the sands of time.

+1 :)

**Review by jfkhoury — APPROVED** [review]
👍  💯

**Pull request merged**